### PR TITLE
Adds support for IREE external parameters at module export time.

### PR DESCRIPTION
**Pull request by stellaraccident**
* Adds kwargs to builtins that create parameters and globals:
  * external=True : Emit as an external parameter
  * external_scope : Set an explicit external scope
  * name_mapper : Callback to map logical name on the module to parameter name

Requires an IREE nightly bump.

### CODE REVIEW

File: tests/aot/globals_test.py [hunk 1]
```diff
@@ -242,9 +216,7 @@ def testTensorUpdateGlobalReturnNone(self):
         update_example = torch.randn(1, 1, 4)
 
         class UpdateState(CompiledModule):
-            state0 = export_global(
-                state_example, mutable=True, initialize=False
-            )
+            state0 = export_global(state_example, mutable=True, initialize=False)
```
Comment (dan-garvey) [inline tests/aot/globals_test.py:219]: looks like we're using different black commands? Probably need a contributing.md

**Review by dan-garvey — APPROVED** [review]
lgtm, do we want to start adding e2e tests for stuff like this that include actually loading the parameters, etc? I suppose I should check the roadmap before I ask that

File: tests/aot/globals_test.py (hunk 1 above)
Comment (stellaraccident) [reply tests/aot/globals_test.py:219]: Can you add an issue here. Should probably just do something like what IREE is doing.

**Comment by stellaraccident** [comment]
> lgtm, do we want to start adding e2e tests for stuff like this that include actually loading the parameters, etc? I suppose I should check the roadmap before I ask that

I would like to, yes. Just getting the base infra in first -- but please do if you feel inspired in the meantime. I'm going to have my hands full for the next day just getting some things straightened outside of that.

**Pull request merged**